### PR TITLE
fix(config): locate the correct `.env` file to use

### PR DESCRIPTION
**Pull request by MarcelGerber**
This fixes the issue that [running the `exportMetadata` script resulted in an error message](https://owid.slack.com/archives/C01D295TM4K/p1616507884021000), and makes the config code more robust.

How this happened: When running `node itsJustJavascript/db/exportMetadata.ts`, dotenv was looking for a config file `itsJustJavascript/.env` - which doesn't exist.
This then meant that `ormconfig.js` simply used the default database settings (for Typeorm), and MySQL threw a not-very-helpful error message.

The fix now is that we're using `pkg-dir` to find our `package.json` file, and always use the `.env` file on that level. This makes it so that our settings code always uses the same config file, no matter whether it's run from `itsJustJavascript` or not.

### CODE REVIEW

File: settings/clientSettings.ts [hunk 1]
```diff
@@ -4,7 +4,18 @@
 // bundled and shipped out to our users.
 
 import dotenv from "dotenv"
-dotenv.config()
+import fs from "fs"
+
+// only run the below code if we're in a node environment
+if (fs.statSync !== undefined) {
```
Comment (MarcelGerber) [inline settings/clientSettings.ts:10]: it's still weird and surprisingly tricky to have some block of code that runs on node, but doesn't run in webpack. or, ideally, wouldn't even be bundled into the webpack bundle.

**Comment by danielgavrilov** [comment]
But if I SSH into the live server and run `node itsJustJavascript/db/exportMetadata.js` it works fine?

Is the problem with `refresh-staging.sh` that it runs `cd itsJustJavascript && node db/exportMetadata.js`?

**Comment by MarcelGerber** [comment]
> But if I SSH into the live server and run `node itsJustJavascript/db/exportMetadata.js` it works fine?
> 
> Is the problem with `refresh-staging.sh` that it runs `cd itsJustJavascript && node db/exportMetadata.js`?

ah yes, then that's probably the case.
however, I think config loading being more robust is a win either way?

**Comment by danielgavrilov** [comment]
Right, yeah it's definitely a win!

Why does it add +11KB to `commons.js`? Is it possible `pkg-dir` gets bundled?

**Comment by MarcelGerber** [comment]
> Is it possible `pkg-dir` gets bundled?

Yes 😭 - and it's hard to fix

**Comment by danielgavrilov** [comment]
How about we just hardcode the directory, relative to `__dirname`?

**Comment by MarcelGerber** [comment]
Okay, I just fell for this again.
Turns out in `refresh-staging.sh`, this line fails:
```sh
cd live/itsJustJavascript && node db/exportMetadata.js --with-passwords /tmp/owid_metadata_with_passwords.sql
```

The script then continues, downloads the _stale_ `owid_metadata_with_passwords` file, and then the refresh is not really a refresh and I'm confused for the next 10 minutes 🤔

**Comment by MarcelGerber** [comment]
Okay, I've updated this now to use a tiny custom function that finds the base dir containing `package.json` - should be better now!

**Review by danielgavrilov — APPROVED** [review]
[Deployed on playfair](https://playfair-owid.netlify.app) to test it – everything seems good! 👍

**Pull request merged**